app/views/user_sessions/_new.html.erb

```diff
@@ -1,6 +1,6 @@
 <div class="container">
-
-<%= form_for :user_session, :as => :user_session, :url => :user_sessions, :html => {:class => "form col-md-6 offset-3"} do |f| %>
+<%= user_sessions_url + "?return_to=" + request.fullpath %>
```

Comment:
Can you please shift this redirecting issue to a new separate PR?
I want to give you credits for this also.